pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+        # instance and modified instance
+        for p in list(modified_instance):
+            if orig_instance[p] == modified_instance[p]:
+                del modified_instance[p]
+
+        # Confirm no key properties in remaining modified instance
+        key_props = [p.name for p in six.itervalues(target_class.properties)
+                     if 'key' in p.qualifiers]
+        for p in key_props:
+            if p in modified_instance:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("ModifyInstance cannot modify key property {0!A}",
+                            p))
+
+        # Remove any properties from modified instance not in the property_list
+        if property_list:
+            for p in list(modified_instance):  # create list before loop
+                if p not in property_list:
+                    del modified_instance[p]
+
+        # Exception if property in instance but not class or types do not
+        # match
+        for pname in modified_instance:
+            if pname not in target_class.properties:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Property {0!A} specified in ModifiedInstance is "
+                            "not exposed by class {1!A} in namespace {2!A}",
+                            pname, target_class.classname, namespace))
+
+            cprop = target_class.properties[pname]
+            iprop = modified_instance.properties[pname]
+            if iprop.is_array != cprop.is_array \
+                    or iprop.type != cprop.type \
+                    or iprop.array_size != cprop.array_size:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Instance and class property name={0!A} type "
+                            "or other attributes do not match: "
+                            "instance={1!A}, class={2!A}",
+                            pname, iprop, cprop))
+            # If case of modified_instance property != case of class property
+            # change the name in the modified_instance
+            if iprop.name != cprop.name:
+                modified_instance.properties[iprop.name].name = cprop.name
+
+        # Modify the value of properties in the repo with those from
+        # modified instance inserted into the original instance
+        orig_instance.update(modified_instance.properties)
+        instance_store.update(modified_instance.path, orig_instance)
+
+        return
+
+    ####################################################################
+    #
+    #   GetInstance server request
+    #
+    ####################################################################
+
+    def GetInstance(self, namespace, InstanceName, LocalOnly=None,
+                    IncludeQualifiers=None, IncludeClassOrigin=None,
+                    PropertyList=None):
```

Comment:
In DSP0200, the INSTANCENAME parameter does not specify a namespace, so it makes sense to have a separate target namespace. However, in this method, the namespace can be specified in the Instancename parameter and therefore is being checked in the implementation, leading to an exception about mismatching namespaces that would never even exist without that redundancy. I suggest to get rid of the redundancy by removing the `namespace` argument and by requiring that the namespace is set in the InstanceName argument.

Reply:
See Comment on previous review item

Reply:
done